packages/ember-htmlbars/lib/compat.js

```diff
@@ -1,12 +1,28 @@
 import Ember from 'ember-metal/core'; // for Handlebars export
+import { deprecate } from 'ember-metal/debug';
 import {
   SafeString,
   escapeExpression
 } from 'ember-htmlbars/utils/string';
 
 var EmberHandlebars = Ember.Handlebars = Ember.Handlebars || {};
 
-EmberHandlebars.SafeString = SafeString;
+EmberHandlebars.SafeString = function(string) {
+  deprecate(
+    'Ember.Handlebars.SafeString is deprecated in favor of Ember.String.htmlSafe',
+    false,
+    {
+      id: 'ember-htmlbars.ember-handlebars-safestring',
+      until: '3.0.0',
+      url: 'http://emberjs.com/deprecations/v2.x#toc_use-ember-string-htmlsafe-over-ember-handlebars-safestring'
+    }
+  );
+
+  SafeString.apply(this, arguments);
+};
+EmberHandlebars.SafeString.prototype = Object.create(SafeString.prototype);
```

Comment:
We could also make `SafeString` a getter, so that just using `Ember.Handlebars.SafeString` issues the deprecation (even if you didn't invoke.  This would help catch `foo instanceof Ember.Handlebars.SafeString` with a deprecation also...